fs/localfs/local_fs_test.go

```diff
@@ -273,29 +272,37 @@ func TestLocalFilesystemPath(t *testing.T) {
 	}
 }
 
-func TestDirPrefix(t *testing.T) {
-	cases := map[string]string{
-		"foo":      "",
-		"/":        "/",
-		"/tmp":     "/",
-		"/tmp/":    "/tmp/",
-		"/tmp/foo": "/tmp/",
+func TestSplitDirPrefix(t *testing.T) {
+	type pair struct {
+		prefix   string
+		basename string
+	}
+
+	cases := map[string]pair{
+		"foo":      {"", "foo"},
+		"/":        {"/", ""},
+		"/tmp":     {"/", "tmp"},
+		"/tmp/":    {"/tmp/", ""},
+		"/tmp/foo": {"/tmp/", "foo"},
 	}
 
-	if runtime.GOOS == "windows" {
-		cases["c:/"] = "c:/"
-		cases["c:\\"] = "c:\\"
-		cases["c:/temp"] = "c:/"
-		cases["c:\\temp"] = "c:\\"
-		cases["c:/temp/orary"] = "c:/temp/"
-		cases["c:\\temp\\orary"] = "c:\\temp\\"
-		cases["c:/temp\\orary"] = "c:/temp\\"
-		cases["c:\\temp/orary"] = "c:\\temp/"
-		cases["\\\\server\\path"] = "\\\\server\\"
-		cases["\\\\server\\path\\subdir"] = "\\\\server\\path\\"
+	if isWindows() {
+		cases["c:/"] = pair{"c:/", ""}
+		cases["c:\\"] = pair{"c:\\", ""}
+		cases["c:/temp"] = pair{"c:/", "temp"}
+		cases["c:\\temp"] = pair{"c:\\", "temp"}
+		cases["c:/temp/orary"] = pair{"c:/temp/", "orary"}
+		cases["c:\\temp\\orary"] = pair{"c:\\temp\\", "orary"}
+		cases["c:/temp\\orary"] = pair{"c:/temp\\", "orary"}
+		cases["c:\\temp/orary"] = pair{"c:\\temp/", "orary"}
+		cases["\\\\server\\path"] = pair{"\\\\server\\", "path"}
+		cases["\\\\server\\path\\"] = pair{"\\\\server\\path\\", ""}
+		cases["\\\\server\\path\\subdir"] = pair{"\\\\server\\path\\", "subdir"}
```

Comment:
nit: does it make sense to use _raw strings_ here for readability?

```suggestion
		cases[`c:/`] = pair{`c:/`, ``}
		cases[`c:\`] = pair{`c:\`, ``}
		cases[`c:/temp`] = pair{`c:/`, `temp`}
		cases[`c:\temp`] = pair{`c:\`, `temp`}
		cases[`c:/temp/orary`] = pair{`c:/temp/`, `orary`}
		cases[`c:\temp\orary`] = pair{`c:\temp\`, `orary`}
		cases[`c:/temp\orary`] = pair{`c:/temp\`, `orary`}
		cases[`c:\temp/orary`] = pair{`c:\temp/`, `orary`}
		cases[`\\server\path`] = pair{`\\server\`, `path`}
		cases[`\\server\path\`] = pair{`\\server\path\`, ``}
		cases[`\\server\path\subdir`] = pair{`\\server\path\`, `subdir`}
```